packages/next/client/index.tsx

```diff
@@ -508,12 +509,13 @@ function renderReactElement(
   const reactEl = fn(shouldHydrate ? markHydrateComplete : markRenderComplete)
   if (process.env.__NEXT_REACT_ROOT) {
     if (!reactRoot) {
-      reactRoot = (ReactDOM as any).createRoot(domEl, {
-        hydrate: shouldHydrate,
-      })
+      // Unlike with createRoot, you don't need a separate root.render() call here
+      reactRoot = (ReactDOM as any).hydrateRoot(domEl, reactEl)
```

Comment:
@huozhi the change you had here previously had a check that also did `createRoot` but `hydrateRoot` already creates a `reactRoot` that can be used later on, allowing the code to be a bit simpler